osbenchmark/aggregator.py

```diff
@@ -0,0 +1,216 @@
+        test_execution.revision = test_exe.revision
+        test_execution.distribution_flavor = test_exe.distribution_flavor
+        test_execution.provision_config_revision = test_exe.provision_config_revision
+
+        return test_execution
+
+    # calculate weighted averages for task metrics
+    def calculate_weighted_average(self, task_metrics: Dict[str, List[Any]], iterations: int) -> Dict[str, Any]:
+        weighted_metrics = {}
+
+        for metric, values in task_metrics.items():
+            weighted_metrics[metric] = {}
+            if isinstance(values[0], dict):
+                for item_key in values[0].keys():
+                    if item_key == 'unit':
+                        weighted_metrics[metric][item_key] = values[0][item_key]
+                    else:
+                        item_values = [value.get(item_key, 0) for value in values]
+                        if iterations > 1:
+                            weighted_sum = sum(value * iterations for value in item_values)
+                            total_iterations = iterations * len(values)
+                            weighted_metrics[metric][item_key] = weighted_sum / total_iterations
+                        else:
+                            weighted_metrics[metric][item_key] = sum(item_values) / len(item_values)
+            else:
+                if iterations > 1:
+                    weighted_sum = sum(value * iterations for value in values)
+                    total_iterations = iterations * len(values)
+                    weighted_metrics[metric] = weighted_sum / total_iterations
+                else:
+                    weighted_metrics[metric] = sum(values) / len(values)
+        return weighted_metrics
+
+    # verify that all test executions have the same workload
+    def compatibility_check(self, test_store) -> None:
+        first_test_execution = test_store.find_by_test_execution_id(list(self.test_executions.keys())[0])
+        workload = first_test_execution.workload
+        for id in self.test_executions.keys():
+            test_execution = test_store.find_by_test_execution_id(id)
+            if test_execution:
+                if test_execution.workload != workload:
+                    raise ValueError(f"Incompatible workload: test {id} has workload '{test_execution.workload}' instead of '{workload}'")
+            else:
+                raise ValueError("Test execution not found: ", id)
+        return True
+
+    # driver code
+    def aggregate(self) -> None:
+        test_execution_store = metrics.test_execution_store(self.config)
+        if self.compatibility_check(test_execution_store):
+            for id in self.test_executions.keys():
+                test_execution = test_execution_store.find_by_test_execution_id(id)
+                if test_execution:
+                    self.config.add(config.Scope.applicationOverride, "workload", "repository.name", "default")
```

Comment:
Repository might not be called "default". We might need another parameter called `--workload-repository` in benchmark.py that we use to initialize the Aggregator class

Reply:
Added this check 👍